src/components/newOrderForm.jsx

```diff
@@ -12,59 +14,108 @@ const NewOrderForm = React.createClass({
 
     getInitialState () {
         return {
-            restaurant: '',
+            restaurant: {
+                text: '',
+                isValid: true
+            },
             deadline: {
-                hour: '',
+                text: '',
                 isValid: true
             },
             deliveryTime: {
-                hour: '',
+                text: '',
+                isValid: true
+            },
+            menu: {
+                text: '',
+                isValid: true
+            },
+            description: {
+                text: ''
+            },
+            password: {
+                text: '',
+                isValid: true
+            },
+            passwordRepeat: {
+                text: '',
+                isValid: true
+            },
+            author: {
+                text: '',
+                isValid: true
+            },
+            deliveryCost: {
+                text: '',
                 isValid: true
             },
-            menu: '',
-            description: '',
-            password: '',
-            passwordRepeat: '',
-            author: '',
-            deliveryCost: '',
-            extraCostPerMeal: ''
+            extraCostPerMeal: {
+                text: '',
+                isValid: true
+            }
         };
     },
 
-    handleHourChange (id, hour) {
+    handleFieldChangeWithValidator (id, value, isValid) {
         this.setState(oldState => {
             oldState[id] = {
-                hour,
-                isValid: this.validateHour(hour)
+                text: value,
+                isValid: isValid
             };
         });
     },
 
-    validateHour (hourInput) {
-        const pattern = /([0-9]{1,2})\:([0-9]{2})/;
-        if (!pattern.test(hourInput)) {
-            return false;
-        }
+    handleHourChange (event) {
+        const {id, value} = event.target;
+        this.handleFieldChangeWithValidator(id, value, validateHour(value));
+    },
 
-        const [, hour, minutes] = pattern.exec(hourInput);
-        if (hour > 24 || minutes > 59) {
-            return false;
-        }
+    handleTextChange (event, isFieldRequired) {
+        const {id, value} = event.target;
+        const isValid = isFieldRequired === true ? validateMinimalLength(value, 1) : true;
+        this.handleFieldChangeWithValidator(id, value, isValid);
+    },
 
-        return true;
+    handleRequiredTextChange (event) {
+        this.handleTextChange(event, true);
+    },
 
+    handleMenuChange (event) {
+        const {id, value} = event.target;
+        const isValid = validateMinimalLength(value, 1) && validateUrl(value);
```

Comment:
We can't as react-bootstrap inputs do not support any of minlength nor pattern attributes. If we would use standard html inputs then that would be great.
There's a react-bootstrap helper method "getInputDOMNode()" to get underneath <input> tag, but I didn't know about it at the moment of writing this code and to be honest I am not a fan of using it. And to be more honest - I think classic bootstrap would be better than this react-shit-strap as it would be easier to use normal html5 features.